mars/services/task/execution/ray/executor.py

```diff
@@ -360,6 +393,23 @@ async def get_progress(self) -> float:
 
     async def cancel(self):
         """Cancel execution."""
+        logger.info("Start to cancel task %s.", self._task)
+        self._cancelled = True
+        if (
+            self._execute_subtask_graph_task is not None
+            and not self._execute_subtask_graph_task.cancelled()
+        ):
+            self._execute_subtask_graph_task.cancel()
+        subtask_num = len(self._output_object_refs_nums)
+        if subtask_num > 0:
+            pos = 0
+            for i in range(0, subtask_num):
+                if i > 0:
+                    pos += self._output_object_refs_nums[i - 1]
```

Comment:
WIll making `_output_object_refs_nums` as a dict(from sutbask to output count) make implementation more simple?

Reply:
Yes, dict is easy to understand. I'll fix it.